PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/runner/functionapp/config/runstate/FunctionAppDeployStateUtil.kt

```diff
@@ -156,7 +156,11 @@ object FunctionAppDeployStateUtil {
 
             functionAppStop(app, processHandler)
 
-            KuduClient.kuduZipDeploy(zipFile, app.publishingProfile, app.name(), processHandler)
+            val kuduBaseUrl = "https://" + app.defaultHostName().toLowerCase()
+                    .replace("http://", "")
+                    .replace(app.name().toLowerCase(), app.name().toLowerCase() + ".scm")
+
+            KuduClient.kuduZipDeploy(zipFile, app.publishingProfile, app.name(), kuduBaseUrl, processHandler)
```

Comment:
Do discuss: since we have same logic to detect a correct base name, maybe create a overrider for `kuduZipDeploy()` that will take an instance of `WebAppBase` and then make this transformation under the hood:

```
fun kuduZipDeploy(zipFile: File, app: WebAppBase, processHandler: RunProcessHandler) {
   ...
}
```

Reply:
Updated.